src/intree.js

```diff
@@ -138,15 +139,21 @@ module.exports.setup = async function({cfg, schemaset}) {
         return slugids[label] = slugid.nice();
       }
     };
-    let errors = validate(config, schema);
+    let version = config.version;
+    
+    let errors;
+    if (version == 0) {
+      errors = validate(config, schema);
+    } else {
+      errors = validate(config, schema.replace(/\.ya?ml$/, '.v1.yaml'));
```

Comment:
So basically, someone calls this function with "schema", and it decides to use a different schema.

I'm glad this got merged, but this would be a good thing to follow up on in a refactor.

Reply:
I agree that a refactor is needed here. It seems like to do this properly I would need to modify `lib-validate`? Brian is suggesting some good ideas there

Reply:
Perhaps - or perhaps just re-thinking the "in-tree" abstraction in general?  What is this function's job, and why do we tell it what schema to expect?  Defining that better might make it clearer how to handle this.